### ISSUE TITLE
Deep ranking

### ISSUE THREAD

**Issue by Yandychang1** [Microsoft/CNTK#2390]
I am looking for an tutorial of image similarity using deep neural networks. Is there one available?

**Comment by sayanpa** [comment]
We do not have any tutorial on this topic specifically yet. But you can create a simple one quickly. You can use an existing pretrained model say VGG (for which there are examples - CNTK 205 / 301) and readout the last but one layer. Now you can train a small model with two inputs for 2 images. Take the trimmed VGG output as the input to the a few Dense (MLP) layers, use Cosine Distance or use cosine distance with negative sampling as the loss. Train your model.

If you want to contribute one, we will be happy to work with you.

**Comment by cha-zhang** [comment]
I would suggest you take a look at our CVPR tutorial slides: 
https://www.cntk.ai/Tutorials/CVPR2017/CVPR_2017_Tutorial_final.pdf 
which has many examples related to vision. 
Recorded video is here: 
https://www.youtube.com/watch?v=lBof7SfWGzU
